packages/ejson/lib/src/decoding.dart

```diff
@@ -32,43 +33,57 @@ const _commonDecoders = {
   Symbol: _decodeSymbol,
   Uint8List: _decodeBinary,
   Uuid: _decodeUuid,
+  DBRef: _decodeDBRef,
   Undefined: _decodeUndefined,
   UndefinedOr: _decodeUndefinedOr,
 };
 
-/// Custom decoders for specific types. Use `register` to add a custom decoder.
-final customDecoders = <Type, Function>{};
-
-final _decoders = () {
+/// Predefined decoders for common types
+final commonDecoders = () {
   // register extra common types on first access
   undefinedOr<T>(dynamic f) => f<UndefinedOr<T>>();
   TypePlus.addFactory(undefinedOr);
   TypePlus.addFactory(<T>(dynamic f) => f<Defined<T>>(), superTypes: [undefinedOr]);
   TypePlus.addFactory(<T>(dynamic f) => f<Undefined<T>>(), superTypes: [undefinedOr]);
+  TypePlus.addFactory(<T>(dynamic f) => f<DBRef<T>>());
   TypePlus.add<BsonKey>();
   TypePlus.add<ObjectId>();
   TypePlus.add<Uint8List>();
   TypePlus.add<Uuid>();
 
-  return CombinedMapView([customDecoders, _commonDecoders]);
+  return _commonDecoders;
 }();
 
+/// Custom decoders for specific types. Use `register` to add a custom decoder.
+final customDecoders = <Type, Function>{};
+
+final _decoders = CombinedMapView([customDecoders, commonDecoders]);
+
 /// Converts [ejson] to type [T].
 ///
+/// [defaultValue] is returned if set, and [ejson] is `null`.
```

Comment:
This doesn't appear to match the implementation - it looks like if the type is nullable, the default value is ignored even if it's set.

Reply:
Yes.. the defaultValue is used for optional, but non-nullable fields. For nullable fields, the default value is always null. I'll elaborate the comment. Check the use in the generated code.

Reply:
Actually, there is no reason to restrict `defaultValue` to be used when `T` is non-nullable. Will adapt code to match comment.